evmdis/main.go

```diff
@@ -5,29 +5,96 @@ import (
 	"fmt"
 	"io/ioutil"
 	"os"
-
-	"github.com/Arachnid/evmdis"
+	"log"
+	"strconv"
+	"flag"
+	"github.com/soad003/evmdis"
 )
 
 func main() {
+	
+	withSwarmHash := flag.Bool("swarm", true, "a bool")
+	ctorMode := flag.Bool("ctor", false, "a bool")
+	logging := flag.Bool("log", false, "a bool")
+	//callsMode := flag.Bool("calls", false, "a bool")
+
+	flag.Parse()
+
+	if !*logging {
+		log.SetOutput(ioutil.Discard)
+	}
+
 	hexdata, err := ioutil.ReadAll(os.Stdin)
 	if err != nil {
 		panic(fmt.Sprintf("Could not read from stdin: %v", err))
 	}
 
-	bytecode := make([]byte, hex.DecodedLen(len(hexdata)))
+	bytecodeLength := uint64(hex.DecodedLen(len(hexdata)))
+	bytecode := make([]byte, bytecodeLength)
+
 	hex.Decode(bytecode, hexdata)
 
-	program := evmdis.NewProgram(bytecode)
-	if err := evmdis.PerformReachingAnalysis(program); err != nil {
-		panic(fmt.Sprintf("Error performing reaching analysis: %v", err))
-	}
-	evmdis.PerformReachesAnalysis(program)
-	evmdis.CreateLabels(program)
-	if err := evmdis.BuildExpressions(program); err != nil {
-		panic(fmt.Sprintf("Error building expressions: %v", err))
+	if bytecode[bytecodeLength-1] == 0x29 &&
```

Comment:
Ok i make them global constants.

Yes it is never reached but it is put in its own basic block (after the last stop) the analysis (data flow, symbolic execution) never reaches it but if you just iterate over the basic blocks for example to print the disassembly you reach this basic block and you likely output missing opcode stuff...